platform/openide.util.ui/src/org/openide/util/Utilities.java

```diff
@@ -1325,6 +1323,25 @@ private static Rectangle findCenterBounds(GraphicsConfiguration gconf, Dimension
         );
     }
 
+    /**
+     * Find the main NetBeans window; must have width or height.
+     * This is used locally to avoid dependency issues. 
+     * @return NetBeans' main window
+     */
+    private static Frame findMainWindow()
+    {
+        Frame f = null;
+        for( Frame f01 : Frame.getFrames() ) {
+            if( "NbMainWindow".equals(f01.getName())) { //NOI18N
```

Comment:
Why doesn't WindowManager.getDefault().getMainWindow() work for this purpose? Is that in a module that is not available from here?

Reply:
> Why doesn't WindowManager.getDefault().getMainWindow() work for this purpose?

Since I didn't write the original code, I can only guess. BTW, the bug report that prompted the change is at https://bz.apache.org/netbeans/show_bug.cgi?id=217737 and the change was made by the window system guru at the time, AFAIK. I looked through emi's old repo, but didn't find the original commit.

I thought it was at least partly a module dependency issue, but I could be wrong. Side effects is another consideration. `Utilities.findMainWindow` looks for an existing window, may return null. `WindowManager...getMainWindow()` creates the window if not found.

Reply:
@eirikbakke Found it (was looking at the wrong Utilities)
https://github.com/emilianbold/netbeans-releases/commit/62e3746e8c1d0e7d3c033291bcc64c8dc8b075d8

But I saw no information about the change.

Reply:
Yes, `openide.windows` has a dependency on `openide.util.ui`. And theoretically `openide.util.ui` could be used in applications without a window system.

Reply:
IIUC, to summarize things

- `openide.util.ui` can not use `WindowManager.getDefault().getMainWindow()` due to dependencies
- The limited functionality based on finding an existing MainWindow can be accessed from any application using NetBeans modules, if the application names the main window "NbMainWindow"
- Don't want findMainWindow() public API
- Could make it internal API (but I don't know where to put it, or exactly the considerations)
- If, in the future, there is a problem with requiring "NbMainWindow" could handle a `NetBeansPlatformMainWindowName` service (or somesuch e.g. SystemProperty) (I'm tempted to put an emoji here, grimace perhaps?)

Reply:
OK, thanks for the detailed explanation!

Reply:
@eirikbakke hope you enjoyed it ;-) I'm not overly familiar with all the issues involved, especially the innards of focus system (not to mention NBs module system), the details are more for me, and to give those who ask "why?" things to consider. But most of all, for someone to point out flaws in my assumptions/reasoning.